### ISSUE TITLE
Stops training with this error

### ISSUE THREAD

**Issue by shriramhr** [bricewalker/Hey-Jetson#7]
2019-07-19 04:56:06.472125: I tensorflow/stream_executor/dso_loader.cc:153] successfully opened CUDA library libcublas.so.10 locally
3455/3455 [==============================] - 21350s 6s/step - loss: 356.8575 - val_loss: 237.5635
Epoch 2/30
1883/3455 [===============>..............] - ETA: 2:39:53 - loss: nan     Batch 1882: Invalid loss, terminating training

**Comment by bricewalker** [comment]
This is expected behavior. Your error message shows that your loss value is NaN. This line in the train_model() function sets model training to terminate if the value of the output of the loss calculation is nan:

terminator = callbacks.TerminateOnNaN()

It is then referenced with this line in the function:

hist = model.fit_generator(generator=audio_gen.next_train(), steps_per_epoch=steps_per_epoch,

        epochs=epochs, validation_data=audio_gen.next_valid(), validation_steps=validation_steps,

        callbacks=[checkpointer, terminator, logger, time_machiner, tensor_boarder], verbose=verbose)

It looks like your model ran into the issue of exploding/vanishing gradients and the loss value became too large or small of a number. I suggest you read up on this issue. I referenced a few of the techniques I used to address these issues in the provided notebook. This is one of the biggest and most common issues plaguing RNN's so I'm sorry but I won't be able to help you any further.
